Automattic/wp-calypso
# Add New G Suite Cancellation Dialog

#### Changes proposed in this Pull Request

* New Cancelation Survey for G Suite
* Additional features added to `MultipleChoiceQuestion`

##### Screens

**Full Screen**
<img width="300" alt="Full Screen Features" src="https://user-images.githubusercontent.com/2810519/54648921-4a499780-4a65-11e9-867c-2ecf4cc2fbc2.png" ><img width="300" alt="Full Screen Survey" src="https://user-images.githubusercontent.com/2810519/54562351-121e5800-4984-11e9-9b03-05bb344d0a3c.png" ><img width="300" alt="Full Screen Survey Selected" src="https://user-images.githubusercontent.com/2810519/54562411-2f532680-4984-11e9-8f5b-2f853700f8f0.png" >

**Mobile**
<img width="300" alt="Mobile Features" src="https://user-images.githubusercontent.com/2810519/54648929-56cdf000-4a65-11e9-934a-1b4e46e57627.png" ><img width="300" alt="Mobile Survey" src="https://user-images.githubusercontent.com/2810519/54562439-3bd77f00-4984-11e9-95d1-b313e9b3105a.png" >

##### Next Steps
We would like to have a button to open happy chat with the "I was unable to activate or use it" survey option, but we should roll that functionality out in an A/B test after the initial form.

##### Tracks Events

**Features Step**
![features_tracks](https://user-images.githubusercontent.com/2810519/54562501-60cbf200-4984-11e9-88f2-e65cc747af9e.png)

1. `calypso_purchases_gsuite_remove_purchase_features_view` on viewing the screen/step
2. `calypso_purchases_gsuite_remove_purchase_keep_it_click` on clicking "I'll Keep It"
3. `calypso_purchases_gsuite_remove_purchase_next_step_click` on clicking "Next Step"
    * `step`: the we are currently on
    *  `next_step` the step we are transitioning to
4. `calypso_domain_management_gsuite_learn_more_click'` on click the "Learn More" link. Provided by the `GSuitePurchaseFeatures` Component

**Survey Step**
![survey_tracks](https://user-images.githubusercontent.com/2810519/54562504-6295b580-4984-11e9-81b5-5019531ffbd4.png)

1. `calypso_purchases_gsuite_remove_purchase_survey_view`on viewing the screen/step
2. `calypso_purchases_gsuite_remove_purchase_survey_answer_change` when a survey answer is changed.
   * `answer_id`: the id of the answer changed to
3. `calypso_purchases_gsuite_remove_purchase_keep_it_click` on clicking "I'll Keep It"
4. `calypso_purchases_gsuite_remove_purchase_prev_step_click` on clicking "Previous Step"
    * `step`: the we are currently on
    *  `prev_step`: the step we are transitioning to
5. `calypso_purchases_gsuite_remove_purchase_click` on clicking "Remove Now"
   * `answer`: the id of the answer chosen in the survey



#### Testing instructions

1. Have a G Suite Subscription you are willing to remove
1. Activate analytics logging by typing `localStorage.setItem( 'debug', 'calypso:analytics:*' )` into the console
1. As you test, keep an eye on the above tracks events and make sure they are appearing as indicated.
1. Navigate to `/me/purchases/`
1. Find the G Suite Subscription you are willing to cancel and click on it
1. Click "Remove G Suite" <img width="500" alt="G Suite Purchase Screen" src="https://user-images.githubusercontent.com/2810519/54323303-e8e17e80-45b5-11e9-9204-3a19c05e9ed0.png">
1. Confirm that you see the new list of G Suite Features
1. Click "Next Step"
1. Confirm you can click each question
1. Click "Remove Now"
1. Confirm that the buttons and survey options are disabled
1. Confirm you do not get a notice that the survey failed to save
1. Confirm you eventually get a notice that your purchase was removed
